spec/index.md

```diff
@@ -4391,28 +4393,43 @@ Env = {
 }
 
 (
-  (A.mode = upgrade or A.mode = upgrade {skip_pre_upgrade = false})
+  (A.mode = upgrade U and U.skip_pre_upgrade ≠ true)
```

Comment:
What I mean is, assuming A is the textual candid argument, isn't the correct condition (taking account of the defaults) something like
```suggestion
  (A.mode = variant { upgrade = 0 } and 0 = null or (0 = opt U and  U.skip_pre_upgrade ≠ opt true))
```
But that is really ugly...

Reply:
Or is there some convention that let's you finesse this?

Reply:
I assume an informal dot notation on `A` since `A` is turned into a binary payload using the function `candid(A)`.